
vop2_win_data rk3588_vop_win_data[] reordering to:

Esmart0-win0
Esmart1-win0
Esmart2-win0
Esmart3-win0
Cluster0-win0
Cluster1-win0
Cluster2-win0
Cluster3-win0

This makes Kodi working out-of-box with DRM planes rendered HW decoded video playback :-)

diff --speed-large-files --no-dereference --minimal -Naur linux-6.16-rc6/drivers/gpu/drm/rockchip/rockchip_vop2_reg.c linux-6.16-rc6/drivers/gpu/drm/rockchip/rockchip_vop2_reg.c
--- linux-6.16-rc6/drivers/gpu/drm/rockchip/rockchip_vop2_reg.c	2025-07-16 09:06:39.910002451 +0200
+++ linux-6.16-rc6/drivers/gpu/drm/rockchip/rockchip_vop2_reg.c	2025-07-16 08:57:37.502470059 +0200
@@ -1119,6 +1119,74 @@
  */
 static const struct vop2_win_data rk3588_vop_win_data[] = {
 	{
+		.name = "Esmart0-win0",
+		.phys_id = ROCKCHIP_VOP2_ESMART0,
+		.possible_vp_mask = BIT(0) | BIT(1) | BIT(2) | BIT(3),
+		.formats = formats_esmart,
+		.nformats = ARRAY_SIZE(formats_esmart),
+		.format_modifiers = format_modifiers,
+		.base = 0x1800,
+		.layer_sel_id = { 2, 2, 2, 2 },
+		.supported_rotations = DRM_MODE_REFLECT_Y,
+		.type = DRM_PLANE_TYPE_OVERLAY,
+		.axi_bus_id = 0,
+		.axi_yrgb_r_id = 0x0a,
+		.axi_uv_r_id = 0x0b,
+		.max_upscale_factor = 8,
+		.max_downscale_factor = 8,
+		.dly = { 23, 45, 48 },
+	}, {
+		.name = "Esmart1-win0",
+		.phys_id = ROCKCHIP_VOP2_ESMART1,
+		.possible_vp_mask = BIT(0) | BIT(1) | BIT(2) | BIT(3),
+		.formats = formats_esmart,
+		.nformats = ARRAY_SIZE(formats_esmart),
+		.format_modifiers = format_modifiers,
+		.base = 0x1a00,
+		.layer_sel_id = { 3, 3, 3, 3 },
+		.supported_rotations = DRM_MODE_REFLECT_Y,
+		.type = DRM_PLANE_TYPE_OVERLAY,
+		.axi_bus_id = 0,
+		.axi_yrgb_r_id = 0x0c,
+		.axi_uv_r_id = 0x01,
+		.max_upscale_factor = 8,
+		.max_downscale_factor = 8,
+		.dly = { 23, 45, 48 },
+	}, {
+		.name = "Esmart2-win0",
+		.phys_id = ROCKCHIP_VOP2_ESMART2,
+		.base = 0x1c00,
+		.possible_vp_mask = BIT(0) | BIT(1) | BIT(2) | BIT(3),
+		.formats = formats_esmart,
+		.nformats = ARRAY_SIZE(formats_esmart),
+		.format_modifiers = format_modifiers,
+		.layer_sel_id =  { 6, 6, 6, 6 },
+		.supported_rotations = DRM_MODE_REFLECT_Y,
+		.type = DRM_PLANE_TYPE_OVERLAY,
+		.axi_bus_id = 1,
+		.axi_yrgb_r_id = 0x0a,
+		.axi_uv_r_id = 0x0b,
+		.max_upscale_factor = 8,
+		.max_downscale_factor = 8,
+		.dly = { 23, 45, 48 },
+	}, {
+		.name = "Esmart3-win0",
+		.phys_id = ROCKCHIP_VOP2_ESMART3,
+		.possible_vp_mask = BIT(0) | BIT(1) | BIT(2) | BIT(3),
+		.formats = formats_esmart,
+		.nformats = ARRAY_SIZE(formats_esmart),
+		.format_modifiers = format_modifiers,
+		.base = 0x1e00,
+		.layer_sel_id =  { 7, 7, 7, 7 },
+		.supported_rotations = DRM_MODE_REFLECT_Y,
+		.type = DRM_PLANE_TYPE_OVERLAY,
+		.axi_bus_id = 1,
+		.axi_yrgb_r_id = 0x0c,
+		.axi_uv_r_id = 0x0d,
+		.max_upscale_factor = 8,
+		.max_downscale_factor = 8,
+		.dly = { 23, 45, 48 },
+	}, {
 		.name = "Cluster0-win0",
 		.phys_id = ROCKCHIP_VOP2_CLUSTER0,
 		.base = 0x1000,
@@ -1194,74 +1262,6 @@
 		.max_downscale_factor = 4,
 		.dly = { 4, 26, 29 },
 		.feature = WIN_FEATURE_AFBDC | WIN_FEATURE_CLUSTER,
-	}, {
-		.name = "Esmart0-win0",
-		.phys_id = ROCKCHIP_VOP2_ESMART0,
-		.possible_vp_mask = BIT(0) | BIT(1) | BIT(2) | BIT(3),
-		.formats = formats_esmart,
-		.nformats = ARRAY_SIZE(formats_esmart),
-		.format_modifiers = format_modifiers,
-		.base = 0x1800,
-		.layer_sel_id = { 2, 2, 2, 2 },
-		.supported_rotations = DRM_MODE_REFLECT_Y,
-		.type = DRM_PLANE_TYPE_OVERLAY,
-		.axi_bus_id = 0,
-		.axi_yrgb_r_id = 0x0a,
-		.axi_uv_r_id = 0x0b,
-		.max_upscale_factor = 8,
-		.max_downscale_factor = 8,
-		.dly = { 23, 45, 48 },
-	}, {
-		.name = "Esmart1-win0",
-		.phys_id = ROCKCHIP_VOP2_ESMART1,
-		.possible_vp_mask = BIT(0) | BIT(1) | BIT(2) | BIT(3),
-		.formats = formats_esmart,
-		.nformats = ARRAY_SIZE(formats_esmart),
-		.format_modifiers = format_modifiers,
-		.base = 0x1a00,
-		.layer_sel_id = { 3, 3, 3, 3 },
-		.supported_rotations = DRM_MODE_REFLECT_Y,
-		.type = DRM_PLANE_TYPE_OVERLAY,
-		.axi_bus_id = 0,
-		.axi_yrgb_r_id = 0x0c,
-		.axi_uv_r_id = 0x01,
-		.max_upscale_factor = 8,
-		.max_downscale_factor = 8,
-		.dly = { 23, 45, 48 },
-	}, {
-		.name = "Esmart2-win0",
-		.phys_id = ROCKCHIP_VOP2_ESMART2,
-		.base = 0x1c00,
-		.possible_vp_mask = BIT(0) | BIT(1) | BIT(2) | BIT(3),
-		.formats = formats_esmart,
-		.nformats = ARRAY_SIZE(formats_esmart),
-		.format_modifiers = format_modifiers,
-		.layer_sel_id =  { 6, 6, 6, 6 },
-		.supported_rotations = DRM_MODE_REFLECT_Y,
-		.type = DRM_PLANE_TYPE_OVERLAY,
-		.axi_bus_id = 1,
-		.axi_yrgb_r_id = 0x0a,
-		.axi_uv_r_id = 0x0b,
-		.max_upscale_factor = 8,
-		.max_downscale_factor = 8,
-		.dly = { 23, 45, 48 },
-	}, {
-		.name = "Esmart3-win0",
-		.phys_id = ROCKCHIP_VOP2_ESMART3,
-		.possible_vp_mask = BIT(0) | BIT(1) | BIT(2) | BIT(3),
-		.formats = formats_esmart,
-		.nformats = ARRAY_SIZE(formats_esmart),
-		.format_modifiers = format_modifiers,
-		.base = 0x1e00,
-		.layer_sel_id =  { 7, 7, 7, 7 },
-		.supported_rotations = DRM_MODE_REFLECT_Y,
-		.type = DRM_PLANE_TYPE_OVERLAY,
-		.axi_bus_id = 1,
-		.axi_yrgb_r_id = 0x0c,
-		.axi_uv_r_id = 0x0d,
-		.max_upscale_factor = 8,
-		.max_downscale_factor = 8,
-		.dly = { 23, 45, 48 },
 	},
 };
 
